cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -544,41 +529,55 @@ impl ManagedEnvironment {
     }
 }
 
-/// Write a pointer lockfile to the specified `lock_path` storing the
-/// current git revision of the tracked upstream repository identified by a `pointer`.
+/// Write a pointer lockfile to the specified `lock_path`.
+///
+/// The lockfile stores the current git revision of the tracked upstream repository.
+/// When a local revision is specified,
+/// and the local revision is different from the remote revision,
+/// the local revision is also stored in the lockfile.
 ///
 /// When committed to a project, guarantees
 /// that the same version of the linked environment is used by all clones.
+/// When a local revision is specified,
+/// flox will **try to** check out the local revision
+/// rather than the remote revision **failing if it can't**.
 fn write_pointer_lockfile(
-    system: &str,
-    pointer: &ManagedPointer,
-    floxmeta: &FloxmetaV2,
     lock_path: PathBuf,
+    floxmeta: &FloxmetaV2,
+    remote_ref: String,
+    local_ref: Option<String>,
 ) -> Result<GenerationLock, EnvironmentError2> {
-    let local_ref = branch_name(system, pointer, lock_path.parent().unwrap()).unwrap();
-    let remote_branch = remote_branch_name(system, pointer);
-
     let rev = floxmeta
         .git
-        .branch_hash(&remote_branch)
+        .branch_hash(&remote_ref)
         .map_err(ManagedEnvironmentError::GitBranchHash)?;
 
-    let local_rev = match floxmeta.git.branch_hash(&local_ref) {
-        Ok(local_rev) if local_rev == rev => None,
-        Ok(local_ref) => Some(local_ref),
-        // todo: think
-        Err(GitCommandBranchHashError::DoesNotExist) => None,
-        Err(err) => Err(ManagedEnvironmentError::GitBranchHash(err))?,
+    let local_rev = if let Some(ref local_ref) = local_ref {
+        match floxmeta.git.branch_hash(local_ref) {
+            Ok(local_rev) if local_rev == rev => None,
+            Ok(local_rev) => Some(local_rev),
+
+            Err(GitCommandBranchHashError::DoesNotExist) => None,
```

Comment:
nonblocking: now that the caller is passing in a branch name, shouldn't this be an error?

Reply:
I guess, yes, also alignt better with the doc comment